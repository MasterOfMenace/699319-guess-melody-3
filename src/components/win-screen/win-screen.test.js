import React from 'react';
import renderer from 'react-test-renderer';
import {Router} from 'react-router-dom';
import WinScreen from './win-screen.jsx';
import history from '../../history.js';

describe(`Правильное отображение компонента WinScreen`, () => {
  it(`С 3 вопросами и 0 ошибок`, () => {
    const tree = renderer.create(
        <Router
          history={history}
        >
          <WinScreen
            questionsCount={3}
            mistakes={0}
            onReplayButtonClick={()=>{}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it(`С 3 вопросами и 1 ошибкой`, () => {
    const tree = renderer.create(
        <Router
          history={history}
        >
          <WinScreen
            questionsCount={3}
            mistakes={1}
            onReplayButtonClick={()=>{}}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
