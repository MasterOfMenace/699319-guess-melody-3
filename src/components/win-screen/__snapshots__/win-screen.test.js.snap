// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Правильное отображение компонента WinScreen С 3 вопросами и 0 ошибок 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    3
     вопросов и совершили 
    0
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;

exports[`Правильное отображение компонента WinScreen С 3 вопросами и 1 ошибкой 1`] = `
<section
  className="result"
>
  <div
    className="result__logo"
  >
    <img
      alt="Угадай мелодию"
      height="83"
      src="img/melody-logo.png"
      width="186"
    />
  </div>
  <h2
    className="result__title"
  >
    Вы настоящий меломан!
  </h2>
  <p
    className="result__total"
  >
    Вы ответили правильно на 
    2
     вопросов и совершили 
    1
     ошибки
  </p>
  <a
    className="replay"
    href="/"
    onClick={[Function]}
  >
    Сыграть ещё раз
  </a>
</section>
`;
